### PR TITLE
Avro-2862: Fix primitive schemas losing metadata properties

### PR DESCRIPTION
**Pull request by csplugins**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-2862
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Review by KyleSchoonover — CHANGES_REQUESTED** [review]
Please address changes in comments

**Review by KyleSchoonover — APPROVED** [review]
Looks good to me.

**Comment by KyleSchoonover** [comment]
@martin-g Can you review?

File: lang/csharp/src/apache/test/Schema/SchemaTests.cs [hunk 1]
```diff
@@ -167,6 +167,49 @@ private static void testToString(Schema sc)
...
+        [TestCase("{ \"type\": \"boolean\", \"metafield\": \"abc\" }", Schema.Type.Boolean)]
+        [TestCase("{ \"type\": \"int\", \"metafield\": \"abc\" }", Schema.Type.Int)]
+        [TestCase("{ \"type\": \"long\", \"metafield\": \"abc\" }", Schema.Type.Long)]
+        [TestCase("{ \"type\": \"float\", \"metafield\": \"abc\" }", Schema.Type.Float)]
+        [TestCase("{ \"type\": \"double\", \"metafield\": \"abc\" }", Schema.Type.Double)]
+        [TestCase("{ \"type\": \"bytes\", \"metafield\": \"abc\" }", Schema.Type.Bytes)]
+        [TestCase("{ \"type\": \"string\", \"metafield\": \"abc\" }", Schema.Type.String)]
+        public void TestPrimitiveWithMetadata(string s, Schema.Type type)
+        {
+            Schema sc1 = Schema.Parse(s);
```
Comment (martin-g) [inline lang/csharp/src/apache/test/Schema/SchemaTests.cs:180]: Please use better names for the variables than `s`, `sc1`, `rc`, etc.

**Review by martin-g — CHANGES_REQUESTED** [review]

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you for the contribution, @csplugins !

**Pull request merged**